File: C++/Problems/Problems/problem_3760.cpp
#include "problem_3760.h"

bool problem_3760::test() {
    std::string s = "abab";

    int expected = 2;

    auto result = maxDistinct(s);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_3760::maxDistinct(std::string s) {
    std::vector seen(26, false);
    int result = 0;
    for (auto c : s)
    {
        int id = c - 'a';
        if (!seen[id])
            result++;
        seen[id] = true;
    }

    return result;
}

File: Python/Problems/Problems/problem3760.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3760(Problem):
    def test(self) -> bool:
        s = "abab"

        expected = 2

        result = self.maxDistinct(s)

        print(result)

        return result == expected

    def maxDistinct(self, s: str) -> int:
        return len(set(s))